src/java.base/share/classes/jdk/internal/foreign/SegmentBulkOperations.java

```diff
@@ -378,6 +378,198 @@ private static long vectorizedMismatchLargeForBytes(MemorySessionImpl aSession,
         return ~remaining;
     }
 
+    //private static final HexFormat HF = HexFormat.ofDelimiter(" ");
+
+    /**
+     * {@return the shortest distance beginning at the provided {@code fromOffset}
+     *          to the encountering of a zero byte in the provided {@code segment}
+     *          checking bytes before the {@code toOffset}}
+     * <p>
+     * The method is using a heuristic method to determine if a long word contains a
+     * zero byte. The method might have false positives but never false negatives.
+     * <p>
+     * This method is inspired by the `glibc/string/strlen.c` implementation
+     *
+     * @param segment    to examine
+     * @param fromOffset from where examination shall begin (inclusive)
+     * @param toOffset   to where examination shall end (exclusive)
+     * @throws IllegalArgumentException if the examined region contains no zero bytes
+     *                                  within a length that can be accepted by a String
+     */
+    @ForceInline
+    public static int strlenByte(final AbstractMemorySegmentImpl segment,
```

Comment:
Can we put these back in `StringSupport`? It makes it hard to review the differences side-by-side. (We can maybe decide later whether to move). In general if we keep the `StringSupport` class it's a bit weird to have the `strlen` methods not be there -- but I agree there's a similarity between this and other bulk operations.